File: frontend/src/features/actividades/validar_integracion_documentos/models/validar_integracion_documentos.ts
import { ValidarInformacionBBVA } from "../../validar_informacion/models/validar_informacion";

export interface ValidarIntegracionDocumentos {
  id_expediente: number;
  encabezado: ValidarIntegracionEncabezado;
  formulario: ValidarIntegracionDocumentosFormulario;
  validar_informacion_data: ValidarInformacionBBVA
}

export interface GuardarValidarIntegracionDocumentos {
  formulario: ValidarIntegracionDocumentosFormulario;
  validar_informacion_data: ValidarInformacionBBVA
}

export interface ValidarIntegracionDocumentosFormulario {
  id: number;
  id_expediente: number;
  id_actividad: string;
  documentos_correctos: boolean | null;
  credito_condicionado: boolean;
  motivo_devolucion: string;
  observaciones: string;
}

export interface ValidarIntegracionEncabezado {
  scoring?: string | null;
  id_tipo_sub_producto?: string | null;
  monto_otorgado_original?: number | null;
  plazo_meses?: number | null;
  tasa?: number | null;
  conditions_organismo_decisor?: string | null;
  codigo_oficina?: string | null;
  descripcion_oficina?: string | null;
  codigo_asesor?: string | null;
  correo_declarativo_original?: string | null;
  telefono_declarativo_original?: string | null;
}

File: backend/Multibanca.Domain.Models/Multibanca/BBVA/ValidarIntegracionDocumentos/validar_integracion_documentos.cs
using Common.Domain.Models;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Multibanca.Domain.Models.Multibanca.BBVA.ValidarIntegracionDocumentos
{
    public class validar_integracion_documentos : base_auditoria
    {
        public int id { get; set; }
        public long id_expediente { get; set; }
        public string id_actividad { get; set; } = "ACT_VALIDAR_INTEGRACION";
        public bool? documentos_correctos { get; set; }
        public bool credito_condicionado { get; set; } = false;
        public string? motivo_devolucion { get; set; }
        public string? observaciones { get; set; }
    }
}
